### ISSUE TITLE
E3031 with AWS Managed Policies in If function

### ISSUE THREAD

**Issue by francilioaraujo** [aws-cloudformation/cfn-lint#4349]
### CloudFormation Lint Version

cfn-lint 1.43.4

### What operating system are you using?

Debian 13

### Describe the bug

AWS Managed Policies from this [list](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/internal/data/aws_managed_policies.json) are accepted, as one can see in the [documentation](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html#sam-function-policies).

Managed policies used in ifs are validated as ARNs.

```
E3031 'CloudWatchLambdaInsightsExecutionRolePolicy' does not match 'arn:(aws[a-zA-Z-]*)?:iam::(\\d{12}|aws):policy/[a-zA-Z_0-9+=,.@\\-_/]+'
template.yaml:17:24
```

### Expected behavior

The validation should pass

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform: AWS::Serverless-2016-10-31

Conditions:
  IsTrue: true

Resources:
  ExampleFunction:
    Type: AWS::Serverless::Function
    Properties:
      Runtime: python3.12
      Handler: handler
      InlineCode: >
        def handler():
          pass
      Policies:
        - !If [IsTrue, CloudWatchLambdaInsightsExecutionRolePolicy, !Ref AWS::NoValue] # This line does not pass
        - Fn::If:                                                                      #
          - IsTrue                                                                     #
          - AWSLambdaExecute                                                           # This line does not pass
          - !Ref AWS::NoValue                                                          #
        - CloudWatchLambdaInsightsExecutionRolePolicy                                  # This line pass
```

**Comment by kddejong** [comment]
Fixed in `aws-sam-translator` 1.108.0. The local SAM transform now correctly resolves short managed policy names to full ARNs inside `Fn::If`. Update with `pip install --upgrade aws-sam-translator`. The maintenance workflow has also been fixed (#4416) to properly bump the `aws-sam-translator` version floor going forward.
